src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt

```diff
@@ -141,6 +143,41 @@ class AdjudicationsService(
       throw e
     }
   }
+  suspend fun updateAdjudicationEvidence(evidenceUpdateEvent: AdjudicationEvidenceUpdateEvent) {
+    val chargeNumber = evidenceUpdateEvent.additionalInformation.chargeNumber
+    val prisonId: String = evidenceUpdateEvent.additionalInformation.prisonId
+    val offenderNo: String = evidenceUpdateEvent.additionalInformation.prisonerNumber
+    val telemetryMap = mutableMapOf(
+      "chargeNumber" to chargeNumber,
+      "prisonId" to prisonId,
+      "offenderNo" to offenderNo,
+    )
+
+    runCatching {
+      val mapping =
+        adjudicationMappingService.getMappingGivenChargeNumber(chargeNumber)
+          .also {
+            telemetryMap["adjudicationNumber"] = it.adjudicationNumber.toString()
+            telemetryMap["chargeSequence"] = it.chargeSequence.toString()
+          }
+
+      val dpsAdjudication = adjudicationsApiService.getCharge(chargeNumber, prisonId)
+      nomisApiService.updateAdjudicationEvidence(
+        mapping.adjudicationNumber,
+        UpdateEvidenceRequest(evidence = dpsAdjudication.reportedAdjudication.evidence.map { it.toNomisUpdateEvidence() }),
+      ).also {
+        telemetryMap["evidenceCount"] = it.evidence.size.toString()
+      }
+
+      telemetryClient.trackEvent(
+        "adjudication-evidence-updated-success",
+        telemetryMap,
+      )
```

Comment:
is there any benefit of adding a }.onSuccess {  here for consistency?  I'm not myself - just wondering

Reply:
Not really - but I suppose it is nice to have symmetry for the telemetry